### DISCUSSION TITLE
If a form has unsaved changes, prompt before closing via escape key

### DISCUSSION THREAD

**Discussion by chadbaldwin** [home-assistant/frontend#21461] [Dashboard]
This is not technically a bug, so I'm putting it here, though I also feel it's not really a "feature" request, it's more of an improvement request.

Just learned this one the hard way. I was building a custom card for one of my dashboards, I'd been working on it for a while. I went to save and I accidentally hit the escape key and the editor just closed...no prompt to ask if I was sure or that I might lose changes, etc.

I think if someone manually clicks "Cancel" that's fine, don't prompt. But I feel like it's pretty easy to accidentally hit the escape key and then poof, your work is gone.

Generally I hate these types of prompts, but I feel like in this case it's a good time to have one.

**Comment by Ziron** [comment]
This should definetly be considered a bug or serious design flaw!
There is nothing more infuriating than losing a bunch of work, sometimes an hours worth or more, just because an accidental press of the escape key discards everything without confirmation!

Possibly related issue: #14151
